src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -127,522 +102,417 @@ cast n =
 -- If the algorithm is only used to find @gcd(a, b)@ (i.e. @s@ and @t@ will be discarded), @a@ and @b@ can be arbitrary integers.
 --
 eea
-    :: forall n b r
-    .  Symbolic b
-    => EuclideanDomain (UInt n r b)
+    :: forall n c r
+    .  Symbolic c
+    => EuclideanDomain (UInt n r c)
     => KnownNat n
-    => KnownNat (NumberOfRegisters (BaseField b) n r)
-    => AdditiveGroup (UInt n r b)
-    => Eq (Bool b) (UInt n r b)
-    => UInt n r b -> UInt n r b -> (UInt n r b, UInt n r b, UInt n r b)
+    => KnownNat (NumberOfRegisters (BaseField c) n r)
+    => AdditiveGroup (UInt n r c)
+    => Eq (Bool c) (UInt n r c)
+    => UInt n r c -> UInt n r c -> (UInt n r c, UInt n r c, UInt n r c)
 eea a b = eea' 1 a b one zero zero one
     where
         iterations :: Natural
         iterations = value @n * 2 + 1
 
-        eea' :: Natural -> UInt n r b -> UInt n r b -> UInt n r b -> UInt n r b -> UInt n r b -> UInt n r b -> (UInt n r b, UInt n r b, UInt n r b)
+        eea' :: Natural -> UInt n r c -> UInt n r c -> UInt n r c -> UInt n r c -> UInt n r c -> UInt n r c -> (UInt n r c, UInt n r c, UInt n r c)
         eea' iteration oldR r oldS s oldT t
           | iteration Haskell.== iterations = (oldS, oldT, oldR)
-          | otherwise = bool @(Bool b) rec (if Haskell.even iteration then b - oldS else oldS, if Haskell.odd iteration then a - oldT else oldT, oldR) (r == zero)
+          | otherwise = bool @(Bool c) rec (if Haskell.even iteration then b - oldS else oldS, if Haskell.odd iteration then a - oldT else oldT, oldR) (r == zero)
             where
                 quotient = oldR `div` r
 
                 rec = eea' (iteration + 1) r (oldR - quotient * r) s (quotient * s + oldS) t (quotient * t + oldT)
 
 --------------------------------------------------------------------------------
+instance (Symbolic c, KnownNat n, KnownRegisterSize r, Haskell.Num b, FromConstant (c Par1) Natural) => ToConstant (UInt n r c) b where
```

Comment:
It's ok to keep the `ToConstant (UInt n r (Interpreter a)) Natural` instance for now, as we do not have the means to properly generalize it yet

Also we do not have a `FromConstant (c Par1) Natural` instance